views/js/runner/provider/qti.js

```diff
@@ -338,12 +338,34 @@ define([
                                 });
                             })
                             .then(function(results){
-                                if(results.testContext){
-                                    self.setTestContext(results.testContext);
+                                var testContext;
+                                var testMap;
+
+                                if (!_.isArray(results)) {
```

Comment:
The goal of the proxy is to abstract the retrieval of data from the way to retrieve data. 
`results` form shouldn't change based on the proxy implementation. If the proxy send actions by bulk, it has the responsibility to send back the data of the given action. 
IMO, the value of `results` must contain only the values from the given `callItemAction` and shouldn't contain data from previous calls